hooks/pre-commit

```diff
@@ -0,0 +1,2 @@
+if [ "`git rev-parse --show-cdup`" != "" ]; then cd `git rev-parse --show-cdup`; fi
```

Comment:
I don't understand what this check does.

Reply:
Sorry, I should have left a comment inline. All it does is change directories to the root of the Git working tree.

This is necessary because the pre-commit hook runs in whatever directory the user is committing from. So, in order to be able to refer to individual script files (the new ones we're adding in `hooks/pre-commit.d`), we need to be able to assume they're in the root of the repo. (Unless I'm missing something obvious.)

Reply:
I **think** my use of `$0` below will solve that problem. Try it out.

Reply:
You're right, it does. Good to know!